video/sweeper_test.go

```diff
@@ -0,0 +1,77 @@
+package video
+
+import (
+	"math/rand"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSweeperInc(t *testing.T) {
+	s := NewSweeper()
+	ids := [][2]string{{"abc", "asdasda"}, {"cde", "sdagkkj"}}
+
+	var wg sync.WaitGroup
+	for range [100]int{} {
+		go func() {
+			wg.Add(1)
```

Comment:
the Add should happen before the `go func()` to prevent races 

https://stackoverflow.com/a/19208908/182709

Reply:
Good catch, thank you